src/struct.h

```diff
@@ -147,3 +147,10 @@ struct PNM_image_header {
   size_t datastart;
   size_t datalength;
 };
+
+struct QOI_pixel {
```

Comment:
I did consider using four local `uint8_t` variables, or one packed `uint32_t`, but the overall complexity and line count would be higher, as values need to be packed into a lookup table and then unpacked for per-channel manipulation.